core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -331,6 +334,50 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
       }
   }
 
+  private[server] def maybeAugmentSslStorePaths(configProps: Properties, previousConfigProps: Map[String, String]): Unit ={
+    val processedFiles = new mutable.HashSet[String]
+    reconfigurables
+      .filter(reconfigurable => ReloadableFileConfigs.exists(reconfigurable.reconfigurableConfigs.contains))
+        .foreach({
+          case reconfigurable: ListenerReconfigurable =>
+            ReloadableFileConfigs.foreach(configName => {
+              val prefixedName = reconfigurable.listenerName.configPrefix + configName
+              if (!processedFiles.contains(prefixedName) && configProps.containsKey(prefixedName) &&
```

Comment:
@rajinisivaram One case I'm not covering here is that whether we should compare the entire `configProps` with `previousConfigProps` first and skip the augment path when they are the same. Right now I'm only comparing the path change of key/trust store, do you think we should do the comparison on entire config change first?

Reply:
I think the current logic seems reasonable since you can have an update with no changes where we just want to reload the key/trust stores. In that case `configProps` would match `previousConfigProps`.